Added Http request sensors entity-initializers for YAML definition

mbarrientos opened the pull request:
Added support for defining basic HTTP sensors at  YAML which retrieves JSON response from http polling to certain URL.

Also fixed deprecated calls on HttpValueFunctions.

mbarrientos committed:
Added entity initializers for HttpRequestSensors. Polling over a URI and retrieving a single JSON value from the response.

mbarrientos committed:
Fix deprecated calls on "chain" to "Functionals.chain"

andreaturli commented on software/base/src/main/java/brooklyn/entity/software/http/HttpRequestSensor.java:
```diff
@@ -0,0 +1,67 @@
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.logging.Logger;
```
why not org.slf4j.Logger; ?

andreaturli commented on software/base/src/main/java/brooklyn/entity/software/http/HttpRequestSensor.java:
```diff
@@ -0,0 +1,67 @@
+    public void apply(final EntityLocal entity) {
+        super.apply(entity);
+
+        //Duration period = entity.getConfig(SENSOR_PERIOD);
```
please remove commented code

andreaturli commented on software/base/src/main/java/brooklyn/entity/software/http/HttpRequestSensor.java:
```diff
@@ -0,0 +1,67 @@
+
+        HttpPollConfig<String> pollConfig = new HttpPollConfig<String>(sensor)
+                //.checkSuccess(HttpValueFunctions.responseCodeEquals(200))
+                //.onFailureOrException(Functions.constant((String)null))
```
same as above

andreaturli commented on software/base/src/test/java/brooklyn/entity/software/HttpRequestSensorTest.java:
```diff
@@ -0,0 +1,74 @@
+
+/**
+ * Created by MBarrientos on 01/07/2014.
+ */
```
we are trying to remove all @author from file. If we get used to the apache-way it will be easier to merge back on `incubator-brooklyn` this contributions

andreaturli commented on software/base/src/test/java/brooklyn/entity/software/HttpRequestSensorTest.java:
```diff
@@ -0,0 +1,74 @@
+        });
+
+        String val = entity.getAttribute(SENSOR_STRING);
+        Assert.assertEquals(val, "myValue", "val=" + val);
```
I prefer static import for things like Assert or Preconditions, but not a strong opinion

andreaturli commented:
Thanks @mbarrientos 

it looks good only few (minor) comments

mbarrientos committed:
Fixed commented code and unused imports

mbarrientos committed:
Changed asserts to static imports

mbarrientos commented:
Thanks @andreaturli , fixed now -> Quite embarrasing unexperienced mistakes... =D

ahgittin commented on software/base/src/main/java/brooklyn/entity/software/http/HttpRequestSensor.java:
```diff
@@ -0,0 +1,62 @@
+
+    private static final org.slf4j.Logger log = LoggerFactory.getLogger(HttpRequestSensor.class);
+
+    public static final ConfigKey<String> SENSOR_JSON_OBJECT = ConfigKeys.newStringConfigKey("JSONObject");
```
for consistency use `jsonObject` - camel case treating abbreviations as words (to prevent abominations like `RESTHTTPURL` !)

ahgittin replied:
`jsonPath` might be a better name, with a comment that (i think) only paths 1 level deep are supported presently.  we really should support the usual json path syntax (cf HTTP PATCH) in time!

wdyt?

ahgittin commented:
@mbarrientos this looks great.  one more minor comment -- really documenting a limitation of brooklyn at present rather than on your end.  (awesome if you want to add support for json paths -- cf [1]).

could you PR this against apache/incubator-brooklyn instead?  we'd like to have it there!  (you need to fill out an ICLA as described at [2], but we are pretty quick at accepting PRs).  it should make managing versions easier if we can avoid this (seaclouds) repo being too different.

[1] http://code.google.com/p/json-path/
[2] https://brooklyn.incubator.apache.org/community/how-to-contribute.html

mbarrientos commented:
Great idea @ahgittin, i didn't know about that library. I'm working on a fix to integrate full json paths, i'll upload changes in a while

mbarrientos committed:
Fixed variable names to fit standard notation

mbarrientos committed:
Add support for json-path (https://code.google.com/p/json-path) for full access to json response

mbarrientos committed:
Moved test to new package

mbarrientos committed:
Refactoring to avoid removing functionality:
now it supports both full path with JSONPath (through HttpValueFunctions.jsonContentsFromPath) and 1-level json element only.

adriannieto commented:
Great addition, works for me.
